kratos/tests/cpp_tests/sparse_matrices/test_sparse_graph_serial.cpp

```diff
@@ -308,10 +311,13 @@ KRATOS_TEST_CASE_IN_SUITE(GraphConstruction, KratosCoreFastSuite)
     SparseTestingInternals::CheckGraph(Agraph, reference_A_map);
 
     //check exporting
-    vector<SparseTestingInternals::IndexType> row_index, col_index;
-    Agraph.ExportCSRArrays(row_index, col_index);
+    span<SparseTestingInternals::IndexType> row_index_span;
+    span<SparseTestingInternals::IndexType> col_index_span;
+    Agraph.ExportCSRArrays(row_index_span, col_index_span);
 
-    SparseTestingInternals::CheckCSRGraphArrays(row_index, col_index, reference_A_map);
+    SparseTestingInternals::CheckCSRGraphArrays(row_index_span, col_index_span, reference_A_map);
+    delete [] row_index_span.begin();
+    delete [] col_index_span.begin();
```

Comment:
so this is related to the spans?
I thought they don't own / allocate memory?

Reply:
note that i don't love this way of using. This is the reason for which i also created a version on which one explicitly needs to pass raw pointers to the function